scripts/processing/segment_sentences.py

```diff
@@ -55,77 +100,34 @@ def load_articles(self):
             it = iter(data)
+            it_name = iter(names)
             for i in range(0, len(data), size):
-                yield {k: data[k] for k in islice(it, size)}
-
-        if use_multiprocessing == 'manager':
-            pass
-
-            time1 = time.time()
-            manager = multiprocessing.Manager()
-            return_dict = manager.dict()
-            jobs = []
-            n_processes = self.segmenting_num_worker    # in addition to the main process, total = n_proc+1
-
-            def work(articles, return_dict, rank):
-                print('process {} needs to segment {} articles'.format(rank,len(articles)))
-                sentences = {}
-                for i, article in enumerate(articles):
-                    sentences[article] = segmenter.segment_string(articles[article])
-
-                    if i % 100 == 0:
-                        print('process {} finish article {}'.format(rank,i))
-
-                print('process {} finish segment'.format(rank))
-                return_dict.update(sentences)
-            rank = 0
-            for item in chunks(self.articles, len(self.articles)//(n_processes)+1):
-                p = multiprocessing.Process(target=work, args=(item, return_dict,rank))
-                rank+=1
-                # Busy wait
-                while len(jobs) >= n_processes:
-                    pass
-
-                jobs.append(p)
-                p.start()
-
-            for proc in jobs:
-                proc.join()
-            self.sentences.update(return_dict)
-
-            time2 = time.time()
-
-            print('total segment spend  spend:', time2-time1)
-        elif use_multiprocessing == 'queue':
-            work_queue = multiprocessing.Queue()
-            jobs = []
-
-            for item in chunks(self.articles, len(self.articles)):
-                pass
-
-        else:
-            # serial option
-            time1= time.time()
-            for i, article in enumerate(self.articles):
-                self.sentences[i] = segmenter.segment_string(self.articles[article])
-
-                if i % 100 == 0:
-                    print('Segmenting article', i)
-            time2 = time.time()
-            print('segmentation spend:', time2-time1)
-        print('End: Sentence Segmentation')
+                yield ({k: data[k] for k in islice(it, size)}, [p for p in islice(it_name, name_size)])
+
+
+
+        n_processes = self.segmenting_num_worker  # in addition to the main process, total = n_proc+1
+        pool = multiprocessing.Pool(n_processes)
+        rank = 0
+        args=[]
+        for item, name_item in chunks(self.articles, self.output_training_files,
+                                      (len(self.articles) - 1) // (n_processes) + 1,
+                                      (len(self.output_training_files) - 1) // (n_processes) + 1):
+            args.append((item, name_item, rank, segmenter))
+            rank += 1

```

Comment:
multiprocessing.map supports automatic chunking. Is there any benefit due to manually chunking the articles here?